smoke-tests/src/test/java/com/splunk/opentelemetry/AppServerTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry;
+
+import com.google.protobuf.ByteString;
+import io.opentelemetry.api.trace.TraceId;
+import io.opentelemetry.proto.trace.v1.Span;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.stream.Collectors;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.junit.jupiter.api.Assertions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AppServerTest extends SmokeTest {
```

Comment:
Please document tests/asserts. They are quite long and complex

Reply:
Every assert has a description, which is also printed out when assertion fails. Should I find better wording for those descriptions or you want some additional kind of documentation?

Reply:
... unfortunately messages come as a last parameter in assertion methods, which makes the code inconvenient to read.

Reply:
I think more about what do these test actually do. Like "First servlet calls the second , which returns all headers (why?)". Something about test setup, et high-ish level

Reply:
Added javadoc describing the expected trace.